src/test/java/com/uid2/admin/vertx/SiteServiceTest.java

```diff
@@ -694,5 +703,176 @@ void addSiteWithDuplicateDomainNames(Vertx vertx, VertxTestContext testContext)
+        fakeAuth(Role.MAINTAINER);
+        setSites();
+        post(vertx, testContext, "api/site/app_names?id=2", "", response -> {
+            assertEquals(400, response.statusCode());
+            assertEquals("must specify a valid site id", response.bodyAsJsonObject().getString("message"));
+            testContext.completeNow();
+        });
+    }
+
+    @Test
+    void appNameBadSiteId(Vertx vertx, VertxTestContext testContext) {
+        fakeAuth(Role.MAINTAINER);
+        setSites();
+        post(vertx, testContext, "api/site/app_names?id=asdf", "", response -> {
+            assertEquals(400, response.statusCode());
+            assertEquals("unable to parse site id For input string: \"asdf\"", response.bodyAsJsonObject().getString("message"));
+            testContext.completeNow();
+        });
+    }
 
+    @Test
+    void appNameNoDomainNames(Vertx vertx, VertxTestContext testContext) {
+        fakeAuth(Role.MAINTAINER);
+        setSites(new Site(123, "name", true));
+        JsonObject reqBody = new JsonObject();
+        post(vertx, testContext, "api/site/app_names?id=123", reqBody.encode(), response -> {
+            assertEquals(400, response.statusCode());
+            assertEquals("required parameters: app_names", response.bodyAsJsonObject().getString("message"));
+            testContext.completeNow();
+        });
+    }
+
+    @Test
+    void appNameEmptyNames(Vertx vertx, VertxTestContext testContext) {
+        fakeAuth(Role.MAINTAINER);
+        Site s = new Site(123, "name", true);
+        setSites(s);
+        JsonObject reqBody = new JsonObject();
+        reqBody.put("app_names", new JsonArray());
+        post(vertx, testContext, "api/site/app_names?id=123", reqBody.encode(), response -> {
+            assertEquals(200, response.statusCode());
+            checkSiteResponse(s, response.bodyAsJsonObject());
+            testContext.completeNow();
+        });
+    }
+
+    @Test
+    void appNameDuplicateAppName(Vertx vertx, VertxTestContext testContext) {
+        fakeAuth(Role.MAINTAINER);
+        Site s = new Site(123, "name", true);
+        setSites(s);
+
+        JsonObject reqBody = new JsonObject();
+        JsonArray names = new JsonArray();
+        names.add("abc");
+        names.add("abc");
+        reqBody.put("app_names", names);
+
+        post(vertx, testContext, "api/site/app_names?id=123", reqBody.encode(), response -> {
+            assertEquals(400, response.statusCode());
+            assertEquals("duplicate app_names not permitted", response.bodyAsJsonObject().getString("message"));
+            testContext.completeNow();
+        });
+    }
+
+    @Test
+    void appNameMultipleAppName(Vertx vertx, VertxTestContext testContext) {
+        fakeAuth(Role.MAINTAINER);
+        Site s = new Site(123, "name", true);
+        setSites(s);
+
+        JsonObject reqBody = new JsonObject();
+        JsonArray names = new JsonArray();
+        names.add("abc1");
```

Comment:
for this test, please change the app names used here to be a mix of lower/upper cases to add extra test coverage on the case sensitivity for app names

Reply:
I changed the app names.